chainerx_cc/chainerx/python/routines.cc

```diff
@@ -275,6 +275,20 @@ void InitChainerxCreation(pybind11::module& m) {
           "endpoint"_a = true,
           "dtype"_a = nullptr,
           "device"_a = nullptr);
+    m.def("tri",
+          [](int64_t n, absl::optional<int64_t> m, int64_t k, py::handle dtype, py::handle device) {
+              if (!m.has_value()) {
+                  m = n;
+              }
+              return MoveArrayBody(Tri(n, m.value(), k, GetDtype(dtype), GetDevice(device)));
+          },
+          "N"_a,
+          "M"_a = nullptr,
+          "k"_a = 0,
+          "dtype"_a = "float64",
```

Comment:
Is there a particular reason why we cannot use fp32? That's currently our "default dtype" as seen in e.g. `GetMathResultType`.

Reply:
No particular reason. Now I know that.